### PR TITLE
Add `BaseUser.banner` for all subclasses to access new banners.

### PR DESCRIPTION
**Pull request by AbstractUmbra**
## Summary

The PR adds the access of `BaseUser.banner` as [Discord have mentioned that bots will be able to access this via the usual User payload](https://github.com/discord/discord-api-docs/issues/3095#issuecomment-868817186).

Currently this is untestable as it is not yet released. However the code runs as designed if it were accessible.

I will update this PR when it is formally released.

## Checklist

- [ ] If code changes were made then they have been tested.
    - [x] I have updated the documentation to reflect the changes.
- [ ] This PR fixes an issue.
- [x] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: discord/asset.py [hunk 1]
```diff
@@ -224,6 +224,17 @@ def _from_sticker(cls, state, sticker_id: int, sticker_hash: str) -> Asset:
             animated=False,
         )
 
+    @classmethod
+    def _from_user_bannner(cls, state, user_id: int, banner_hash: str) -> Asset:
```
Comment (NCPlayz) [inline discord/asset.py:228]:
```suggestion
    def _from_user_banner(cls, state, user_id: int, banner_hash: str) -> Asset:
```

File: discord/user.py [hunk 2]
```diff
@@ -123,6 +125,16 @@ def default_avatar(self):
...
 
+    @property
+    def banner(self):
+        """Optional[:class:`Asset`]: Returns the user's banner asset, if available.
+
+        .. versionadded:: 2.0
+        """
+        if self._banner is None:
+            return None
+        return Asset._from_user_bannner(self._state, self.id, self._banner)
```
Comment (NCPlayz) [inline discord/user.py:136]:
```suggestion
        return Asset._from_user_banner(self._state, self.id, self._banner)
```

**Review by NCPlayz — CHANGES_REQUESTED** [review]

File: discord/user.py [hunk 3]
```diff
@@ -123,6 +125,16 @@ def default_avatar(self):
         """:class:`Asset`: Returns the default avatar for a given user. This is calculated by the user's discriminator."""
         return Asset._from_default_avatar(self._state, int(self.discriminator) % len(DefaultAvatar))
 
+    @property
+    def banner(self):
```
Comment (Gobot1234) [inline discord/user.py:129]:
```suggestion
    def banner(self) -> Optional[Asset]:
```

Comment (AbstractUmbra) [reply discord/user.py:129]: As the rest of the properties (i.e `avatar`) had no return annotations, I opted out of typing up these ones.

Comment (bijij) [reply discord/user.py:129]: That's just because user isn't typed yet.
Feel free to annotate

**Comment by AbstractUmbra** [comment]
This is now ready for review. but as discussed within the Discord server, the API side of things is still not finalised Discord side, so changes may be required.

File: discord/user.py [hunk 4]
```diff
@@ -78,6 +78,8 @@ def _update(self, data):
         self.id = int(data['id'])
         self.discriminator = data['discriminator']
         self._avatar = data['avatar']
+        self._banner = data.get('banner', None)
+        self._accent_colour = data.get("accent_color", None)
```
Comment (Rapptz) [inline discord/user.py:82]:
```suggestion
        self._accent_colour = data.get('accent_color', None)
```

File: discord/user.py [hunk 5]
```diff
@@ -127,6 +131,50 @@ def default_avatar(self):
...
+    def accent_colour(self) -> Optional[Colour]:
+        """Optional[:class:`Colour`]: Returns the user's accent colour, if applicable.
+
+        There is an alias for this named :attr:`accent_color`.
+
+        .. versionadded:: 2.0
+
+
+        .. note::
+            This information is only available via :meth:`Client.fetch_user`.
```
Comment (Rapptz) [inline discord/user.py:158]:
```suggestion
        .. versionadded:: 2.0

        .. note::

            This information is only available via :meth:`Client.fetch_user`.
```

File: discord/user.py [hunk 6]
```diff
@@ -127,6 +131,50 @@ def default_avatar(self):
...
+    def accent_color(self) -> Optional[Colour]:
+        """Optional[:class:`Colour`]: Returns the user's accent color, if applicable.
+
+        There is an alias for this named :attr:`accent_colour`.
+
+        .. versionadded:: 2.0
+
+
+        .. note::
+            This information is only available via :meth:`Client.fetch_user`.
```
Comment (Rapptz) [inline discord/user.py:174]:
```suggestion
        .. versionadded:: 2.0

        .. note::

            This information is only available via :meth:`Client.fetch_user`.
```

**Review by Rapptz — COMMENTED** [review]
Just some minor nits

**Pull request merged**